include/reflection-cpp/reflection.hpp

```diff
@@ -347,8 +347,32 @@ namespace detail
 #elif defined(_MSC_VER)
 #endif
 
+    template <typename... Ts, typename F>
+    constexpr void enumerate_types(F&& f)
+    {
+        [&f]<auto... Is>(std::index_sequence<Is...>) {
+            (f.template operator()<Ts, Is>(), ...);
+        }(std::index_sequence_for<Ts...> {});
+    }
+
+    template <auto... Xs, typename F>
+    constexpr void for_values(F&& f)
+    {
+        (f.template operator()<Xs>(), ...);
+    }
+
 } // namespace detail
 
+template <auto B, auto E, typename F>
+constexpr void template_for(F&& f)
```

Comment:
hm, i think we should decide on what naming convention to adopt for this API. I'm puzzled. I at first thought that CamelCase might be good here, but I'm also fine with snake_case. In the end, it should be homogenous everywhere, and the .clang-format should then be adapted. What would you prefer?

Reply:
I am okay with using CamelCase, i just wanted to use here snake_case because future syntax (as per https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p1306r2.pdf) will be `template for`

Reply:
oh i absolutely agree. is `template_for` supposed to be private API? Then we should move it into `detail` sub namespace.

Reply:
p.s.: I also wonder how to actually construct Inspect() such that it can traverse through nested elements, like vector and map. A very little bit like glaze's json serializer, except that we don't care about json syntax - only human readable dumping.

Reply:
No, i think this is something we can expose, this is compile time iteration over integer_sequence

Reply:
> p.s.: I also wonder how to actually construct Inspect() such that it can traverse through nested elements, like vector and map. A very little bit like glaze's json serializer, except that we don't care about json syntax - only human readable dumping.

I can add some extra test for this and try to implement